backward.hpp

```diff
@@ -587,6 +593,29 @@ template <> struct demangler_impl<system_tag::current_tag> {
 
 struct demangler : public demangler_impl<system_tag::current_tag> {};
 
+// Split a string on the platform's PATH delimiter.  Example: if delimiter
+// is ":" then:
+//   ""              --> []
+//   ":"             --> ["",""]
+//   "::"            --> ["","",""]
+//   "/a/b/c"        --> ["/a/b/c"]
+//   "/a/b/c:/d/e/f" --> ["/a/b/c","/d/e/f"]
+//   etc.
+inline std::vector<std::string> split_source_prefixes(const std::string &s) {
+  std::vector<std::string> out;
+  std::string delimiter = BACKWARD_PATH_DELIMITER;
```

Comment:
This makes an extra little heap allocation/copy for no benefit (ok, probably that for a tiny string like that gcc and clang stl are actually packing small strings inside the string object itself, but this is implementation specific).

You can use BACKWARD_PATH_DELIMITER directly as an argument to s.find() (it accepts char*) and `sizeof(BACKWARD_PATH_DELIMITER) - 1` to find out its size (this works because string literals are of type char[X] with the `"\0"` included).

Technically, the using a macro means the literally string is a new one everywhere. Bu the compiler (linker?) will remove duplication. To leave it less to the compiler magic, you can also use a `const char[] = ` instead of macros. I don't remember the complicated C++ rules for a global constant inside a header that is part of multiple compilation units though. Maybe you need to add some `inline` qualifier in there.

If you don't want to bother looking through this, let me know, and I will merge this PR as is. C++ is quite a pain.

Reply:
Done; I have removed the macro and just used `const char[]` for the delimiter, and put it into the `backward::details` namespace.

A global array declared `const` will have internal linkage, see here:

  https://stackoverflow.com/questions/48636462/c-global-const-array-is-it-guaranteed-to-be-merged-optimized-into-one-copy

so there may be multiple copies of the string in different translation units, but they will all be identical in content, so there will be no conflict so long as no one compares the addresses of two of them between translation units (which won't happen in our case).

Starting in C++17 we can use the `inline` keyword on globals variables (along with `std::string_view`) like so:

```
inline constexpr std::string_view kBackwardPathDelimiter = ":";
```

which is the recommended way to declare global string constants in headers starting with C++17, but I know that backward-cpp needs to support older standards, so I guess we cannot do this.